frontend/src/lib/StaticGraphParser.ts

```diff
@@ -263,3 +264,60 @@ export function createGraph(workflow: Workflow): dagre.graphlib.Graph {
 
   return graph;
 }
+
+/**
+ * Perform a transitive reduction over the input graph.
+ *
+ * From [1]: Transitive reduction of a directed graph D is another directed
+ * graph with the same vertices and as few edges as possible, such that for all
+ * pairs of vertices v, w a (directed) path from v to w in D exists if and only
+ * if such a path exists in the reduction
+ *
+ * The current implementation has a time complexity bound `O(n*m)`, where `n`
+ * are the nodes and `m` are the edges of the input graph.
+ *
+ * [1]: https://en.wikipedia.org/wiki/Transitive_reduction
+ *
+ * @param graph The dagre graph object
+ */
+export function transitiveReduction(graph: dagre.graphlib.Graph) {
+  // safeguard against too big graphs
+  if (graph.edgeCount() > 1000 || graph.nodeCount() > 1000) {
```

Comment:
Can we check if graph is null here?

Reply:
For clarification, we enabled strict nullity in typescript config, so when type does not include undefined/null, it cannot be.

Reply:
Thank you Yuan for clarification!